### ISSUE TITLE
POSSIBLE BUG: Auth Token Doesn't Work In Production But Only Works In Development

### ISSUE THREAD

**Issue by jrpolidario** [chrismccord/sync#201]
## My Setup: ##
Rails 4.2, ruby 2.2, sync 0.3.5.

## Summary: ##
1. I had an issue yesterday in that turbolinks is required for sync to work when page is viewed remotely (i.e. via Iphone) with a local env=Development machine I set up. (although it works in when viewing the page in my own local machine). Unfortunately, I couldn't get yet a time to inspect why this happens.

2. However, another issue when I now try to deploy in env=Production to an AWS EC2 Instance is that -- the `config.yml.production.auth_token` doesn't seem to work. I found out this didn't work by trial-and-error when I replaced `production.auth_token` by the `development.auth_token` which is `DEVELOPMENT_SECRET_TOKEN`, and voila the page with sync partial now finally updates (Although now I cant add my own Production token).

I made  sure that I was running:
```bash
bundle exec unicorn -D -c config/unicorn.rb -E production # (THIS IS surely in production)
bundle exec rackup sync.ru -E production --host 0.0.0.0 # (THIS IS surely in production as well)
```
I even ran both
```bash
bundle exec rake tmp:clear
bundle exec rake assets:precompile
```
But the page doesn't still get websocket updates (to render partials). Although, I verified that the websocket is connected with a status 101.

So, I was going crazy haha. But, after trial-and-error, I managed to finally make the production.auth_token work now by running:
```bash
RAILS_ENV=production bundle exec rackup sync.ru -E production --host 0.0.0.0
```
instead of the default:

```bash
bundle exec rackup sync.ru -E production --host 0.0.0.0
```

I thought that the above should already work, but it needed me to append RAILS_ENV=production as well.

Again, I am afraid I do not yet have  the luxury in elaborating the problem nor try to dig into the source code to find out more what is causing this, so I just posted this here in case it might be proved helpful to anyone.

**Comment by MichaelSp** [comment]
Related to #194?

**Comment by ajb** [comment]
This should be closed by #204. Note that 0.4.0 is not yet pushed to rubygems, so you'll have to install from GitHub.
